doc/contributing.md

```diff
@@ -0,0 +1,164 @@
+# How to contribute
+
+* [Discuss your ideas](#discuss-your-ideas)
+* [Clone the repository](#clone-the-repository)
+* [Create your development environment](#create-your-development-environment)
+* [Make your changes](#make-your-changes)
+* [Testing](#testing)
+    * [Run the tests](#run-the-tests)
+    * [Writing your own tests](#writing-your-own-tests)
+* [Open a pull request](#open-a-pull-request)
+* [Optional: Build the documentation](#optional-build-the-documentation)
+* [Optional: Benchmark performance](#optional-benchmark-performance)
+    * [Run the benchmark script](#run-the-benchmark-script)
+    * [Line profiling](#line-profiling)
+    * [Snakeviz visualization](#snakeviz-visualization)
+    * [Memory profiling with memray](#memory-profiling-with-memray)
+
+## Discuss your ideas
+
+We welcome new contributions!
+
+First, discuss your idea by opening a [new issue](https://github.com/computational-cell-analytics/micro-sam/issues/new) in micro-sam.
+
+This allows you to ask questions, and have the current developers make suggestions about the best way to implement your ideas.
+
+## Clone the repository
+
+We use [git](https://git-scm.com/) for version control.
+
+Clone the repository, and checkout the development branch:
+```
+git clone https://github.com/computational-cell-analytics/micro-sam.git
+cd micro-sam
+git checkout dev
+```
+
+## Create your development environment
+
+We use [conda](https://docs.conda.io/en/latest/) to [manage our environments](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html). If you don't have this already, install [miniconda](https://docs.conda.io/projects/miniconda/en/latest/) or [mamba](https://mamba.readthedocs.io/en/latest/) to get started.
+
+Now you can create the environment, install user and develoepr dependencies, and micro-sam as an editable installation:
+```
+conda env create environment-gpu.yml
+conda activate sam
+python -m pip install requirements-dev.txt
+python -m pip install -e .
+```
+
+## Make your changes
+
+Now it's time to make your code changes.
+
+Typically, changes are made branching off from the development branch. Checkout `dev` and then create a new branch to work on your changes.
+```
+git checkout dev
+git checkout -b my-new-feature
+```
+
+You may also find it helpful to look at this [developer guide](development.md), which explains the organization of the micro-sam code.
```

Comment:
Also, will the link to the developer guide page work correctly in the built documentation? I wasn't entirely sure how to build the docs locally with [pdoc](https://pdoc.dev/), so I haven't tested this.

Reply:
Second, should I repeat this sentence up above in the "discuss your ideas" section as well?